src/command.rs

```diff
@@ -246,6 +377,97 @@ impl Run {
 
         args
     }
+
+    /// Callback for `cargo 3ds run`.
+    ///
+    /// This callback handles launching the application via `3dslink`.
+    fn callback(&self, config: &CTRConfig) {
+        // Run the normal "build" callback
+        self.build_args.callback(config);
+
+        eprintln!("Running 3dslink");
+        link(config, self);
+    }
+}
+
+impl Test {
+    /// Callback for `cargo 3ds test`.
+    ///
+    /// This callback handles launching the application via `3dslink`.
+    fn callback(&self, config: &CTRConfig) {
+        if self.no_run {
+            // If the tests don't have to run, use the "build" callback
+            self.run_args.build_args.callback(config)
+        } else {
+            // If the tests have to run, use the "run" callback
+            self.run_args.callback(config)
+        }
+    }
+}
+
+const TOML_CHANGES: &str = "ctru-rs = { git = \"https://github.com/rust3ds/ctru-rs\" }
+
+[package.metadata.cargo-3ds]
+romfs_dir = \"romfs\"
+";
+
+const CUSTOM_MAIN_RS: &str = "use ctru::prelude::*;
+
+fn main() {
+    ctru::use_panic_handler();
+
+    let apt = Apt::new().unwrap();
+    let mut hid = Hid::new().unwrap();
+    let gfx = Gfx::new().unwrap();
+    let _console = Console::new(gfx.top_screen.borrow_mut());
+
+    println!(\"Hello, World!\");
+    println!(\"\\x1b[29;16HPress Start to exit\");
+
+    while apt.main_loop() {
+        gfx.wait_for_vblank();
+
+        hid.scan_input();
+        if hid.keys_down().contains(KeyPad::START) {
+            break;
+        }
+    }
+}
+";
+
+impl New {
+    /// Callback for `cargo 3ds new`.
+    ///
+    /// This callback handles the custom environment modifications when creating a new 3DS project.
+    fn callback(&self) {
+        // Commmit changes to the project only if is meant to be a binary
+        if self.cargo_args.args.contains(&"--lib".to_string()) {
+            return;
```

Comment:
for libs we could probably add the `ctru-rs` dependency but leave everything else alone. Not really saving as much time vs the rest though